File: src/main/java/com/example/demoimdb/model/Rating.java
package com.example.demoimdb.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;
import javax.security.auth.Subject;

@Entity
@Table(name = "rating")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Rating {
    @ManyToOne
    @JoinColumn(name = "account_id")
    @JsonIgnore
    private Account account;
    @ManyToOne
    @JoinColumn(name = "movie_id")
    @JsonIgnore
    private Movie movie;
    @ManyToOne
    @JoinColumn(name = "episode_id")
    @JsonIgnore
    private Episode episode;
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(name = "score")
    private Integer score;
    @Column(name = "date")
    private String date;
}

File: src/entities/Rating.ts
import { Column, Entity, ManyToOne, PrimaryGeneratedColumn } from 'typeorm';
import { Movie } from './Movie';
import { Episode } from './Episode';
import { Account } from './Account';

@Entity({ name: 'ratings' })
export class Rating {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ nullable: false })
  score: number;

  @Column({ nullable: false })
  date: string;

  @ManyToOne(() => Movie, (movie) => movie.ratings, { onDelete: 'SET NULL' })
  movie: Movie;

  @ManyToOne(() => Episode, (episode) => episode.ratings)
  episode: Episode;

  @ManyToOne(() => Account, (account) => account.ratings)
  account: Account;
}
